controllers/openshift_controller.go

```diff
@@ -942,9 +955,15 @@ func (r *KataConfigOpenShiftReconciler) processKataConfigInstallRequest() (ctrl.
 		r.Log.Info("SCNodeRole is: " + machinePool)
 	}
 
+	wasMcJustCreated, err := r.createExtensionMc(machinePool)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, nil
+	} else if wasMcJustCreated {
+		return ctrl.Result{Requeue: true, RequeueAfter: 5 * time.Second}, nil
+	}
 
 	// Create kata-oc MCP only if it's not a converged cluster
-	if machinePool != "master" {
+	if machinePool == "kata-oc" {
```

Comment:
Not sure to understand the rationale behind this change, especially since it is somehow reverted in the next commit.

Reply:
The original `getMcpName()` could return "kata-oc" or "" for a regular cluster or "master" for a converged cluster so the meaning of the conditional was "if this is a regular cluster" (as the comment points out).  The new function is more straightforward, it will only return "kata-oc" or "master" for regular or converged clusters respectively.  Taking this into account, if we want to ask "is this cluster regular" I consider it more straightforward to ask that question directly instead of "is this cluster not non-regular", hence this change.

As for this line being further changed in a subsequent commit, I tried to take a piecemeal approach for this series, gradually improving the code step-by-step to the extent possible at every given stage.  I recognise that this is not the only possible approach but I feared that going from the original code state straight to the final desired state would make for a series with just a couple of fairly complicated commits.

Reply:
Thanks for the clarification.